test/test-integration.js

```diff
@@ -0,0 +1,174 @@
+
+        // Allow loading test resources oudside of test/ directory
+        // (e.g. bower-components)
+        'security.fileuri.strict_origin_policy': false,
+
+        // Enable web components
+        'dom.webcomponents.enabled': true,
+        // Enable touch events
+        'dom.w3c_touch_events.enabled': 1
+      }
+    },
+    desiredCapabilities: {
+      raisesAccessibilityExceptions: true
+    }
+  });
+
+  var dialogs = [
+    { selector: '#d1' },
+    { selector: '#d2' },
+    { selector: '#d3' },
+    { selector: '#d4' }
+  ];
+
+  function openDialog(selector) {
+    client.executeScript(function(selector) {
+      document.querySelector(selector).wrappedJSObject.open();
+    }, [selector]);
+  }
+
+  function closeDialog(selector) {
+    client.executeScript(function(selector) {
+      document.querySelector(selector).wrappedJSObject.close();
+    }, [selector]);
+  }
+
+  function testActions(name, testFn) {
+    test(name, function() {
+      ['click', 'tap'].forEach(testFn);
+    });
+  }
+
+  /**
+   * Perform a marionette operation and assert if an error is thrown.
+   * @param  {Function} testFn operation to perform
+   * @param  {String} message error message for the assert statement
+   */
+  function failOnA11yError(testFn, message) {
+    try {
+      testFn();
+    } catch (err) {
+      // Marionette raises an ElementNotAccessibleError exception when
+      // raisesAccessibilityExceptions is set to true.
+      assert(false, [message, err.message].join(' '));
+    }
+  }
+
+  test('when shown and then hidden, gaia-dialog has correct visibility state ' +
+    'both normally and from the assistive technology standpoint', function() {
+    dialogs.forEach(function(dialog) {
+      openDialog(dialog.selector);
+      failOnA11yError(function() {
+        client.helper.waitForElement(dialog.selector);
+      }, 'gaia-dialog element should be visible both normally and to ' +
+        'assistive technology.');
+
+      closeDialog(dialog.selector);
+      failOnA11yError(function() {
+        client.helper.waitForElementToDisappear(dialog.selector);
+      }, 'gaia-dialog element should be hidden both normally and from ' +
+        'assistive technology.');
+    });
+  });
+
+  suite('gaia-dialog controls, such as buttons are accessible (no error ' +
+    'thrown when clicking and tapping)', function() {
+    testActions('base gaia-dialog controls are accessible', function(action) {
+      var dialog = dialogs[0];
+      openDialog(dialog.selector);
+
+      // Click to close dialog
+      client.helper.waitForElement('button[on-click="close"]')[action]();
+      failOnA11yError(function() {
+        client.helper.waitForElementToDisappear(dialog.selector);
+      }, 'gaia-dialog element should be hidden both normally and from ' +
+        'assistive technology.');
+    });
+
+    testActions('select gaia-dialog controls are accessible', function(action) {
```

Comment:
Simply a wrapper around test that does the same test for different user actions: click and tap